app/src/main/java/shoppinglist/com/shoppinglist/ShoppingListActivity.java

```diff
@@ -0,0 +1,179 @@
+package shoppinglist.com.shoppinglist;
+
+import android.content.Intent;
+import android.location.Location;
+import android.support.v7.app.ActionBarActivity;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.Button;
+import android.widget.ListView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.android.volley.Request;
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.StringRequest;
+import com.google.android.gms.common.ConnectionResult;
+import com.google.android.gms.common.api.GoogleApiClient;
+import com.google.android.gms.location.LocationServices;
+
+import java.util.List;
+
+import shoppinglist.com.shoppinglist.camera.CameraActivity;
+import shoppinglist.com.shoppinglist.database.DatabaseHelper;
+import shoppinglist.com.shoppinglist.database.ShoppingListDatabase;
+import shoppinglist.com.shoppinglist.database.exceptions.PersistingFailedException;
+import shoppinglist.com.shoppinglist.database.orm.ShoppingItem;
+import shoppinglist.com.shoppinglist.database.orm.ShoppingList;
+import shoppinglist.com.shoppinglist.network.NetBlaster;
+
+public class ShoppingListActivity extends ActionBarActivity implements GoogleApiClient.ConnectionCallbacks, GoogleApiClient.OnConnectionFailedListener {
+
+
+    ShoppingList shoppingList = null;
+    ShoppingListDatabase shoppingListDatabase = null;
+    private GoogleApiClient mGoogleApiClient;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        setContentView(R.layout.activity_shopping_list);
+
+
+        //TODO remove singleton or make getInstance api cleaner
+        //this.deleteDatabase(DatabaseHelper.DATABASE_NAME);
+        shoppingListDatabase = DatabaseHelper.getInstance(getApplicationContext());
+
+        try {
+            shoppingList = shoppingListDatabase.getCurrentShoppingList();
+        } catch (PersistingFailedException e) {
+            Toast.makeText(this, "Couldn't get ShoppingList.", Toast.LENGTH_SHORT).show();
+        }
+
+        List<ShoppingItem> items = shoppingList.getItems();
```

Comment:
In the event that `shoppingList = shoppingListDatabase.getCurrentShoppingList()` throws an exception `shoppingList` would probably be `null` causing a crash??? Maybe move everything below the catch that relies on `shoppingList` into the try block?